plugins/outputs/zabbix/lld.go

```diff
@@ -0,0 +1,196 @@
+package zabbix
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/metric"
+)
+
+const (
+	lldName = "lld"
+)
+
+type zabbixLLD struct {
+	log telegraf.Logger
+
+	// cache stores the signature of measurements to avoid processing twice the same.
+	// This signature is generated with the measurement name, tag and tag values
+	cache map[uint64]bool
+
+	// receivedData store LLD values for each host-LDDKey combination
+	receivedData map[hostLLD]lldTags
+
+	// previousReceivedData stores the receivedData of the previous period
+	previousReceivedData map[hostLLD]lldTags
+
+	// lastClear store the time of the last clear of the LLD data
+	lastClear time.Time
+
+	// clearInterval after this number of pushes, all data is considered new.
+	// The idea behind this parameter is to resend known LLDs with low freq in case
+	// previous sent was not processed by Zabbix.
+	clearInterval config.Duration
+}
```

Comment:
Maybe too pedantic but can you please move the struct close to the member-function implementation? Not a must but preferred from my reviewer standpoint. ;-)

Reply:
Fixed in 8bec20fc0 (if I understood correctly)